nix/nixos/cardano-node.nix

```diff
@@ -118,26 +138,25 @@ in {
       description   = "cardano-node node service";
       after         = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-      script = let 
-        # TODO: generate logger config:
-        loggerConfig = cfg.logger.config-file;
-
-        topology = builtins.toFile "topology.json" (builtins.toJSON [{
-          nodeId = cfg.node-id;
-          nodeAddress = {
-            addr = cfg.host;
-            port = builtins.toString cfg.port;
-          };
-          producers = map (p: let s = builtins.split ":" p; in {
-            addr = builtins.head s;
-            port = builtins.head (builtins.tail (builtins.tail s));
-          }) cfg.producers-peers;
-        }]);
-
-      in ''
-        START_TIME=${if (cfg.system-start-time != null) then cfg.system-start-time else "`date \"+%Y-%m-%d 00:00:00\"`"}
-        ${cfg.package}/bin/cardano-node --log-config ${cfg.logger.config-file} --system-start "$START_TIME" --slot-duration ${builtins.toString cfg.slot-duration} node --topology ${topology} --${cfg.consensus-protocol} --node-id ${builtins.toString cfg.node-id} --host ${cfg.host} --port ${builtins.toString cfg.port}
-      '';
+      script = ''
+        GENESIS_START_TIME=$(${pkgs.jq}/bin/jq '.startTime' < ${cfg.genesis-file})
```

Comment:
Is it ok to get the start time defined in genesis file and use it as argument to `--system-start`?

Reply:
Good question. You'll need to pester us for a proper answer on that. I think for mainnet the answer is yes, but for dev and testing I'm not so sure. Ask @mrBliss